### ISSUE TITLE
Timed out while waiting for the machine to boot

### ISSUE THREAD

**Issue by andrewburnheimer** [bbc/nmos-device-connection-management-ri#1]
Followed advice and directions at https://github.com/bbc/nmos-device-connection-management-ri#prerequisites .

Unfortunately, the VM created by Vagrant never seems to boot. (more discussion after snippet)

```
root@server01:/opt/nmos-device-connection-management-ri# /usr/lib/ruby/vendor_ruby/vagrant/exec/vagrant up
You appear to be running Vagrant outside of the official installers.
Note that the installers are what ensure that Vagrant has all required
dependencies, and Vagrant assumes that these dependencies exist. By
running outside of the installer environment, Vagrant may not function
properly. To remove this warning, install Vagrant using one of the
official packages from vagrantup.com.

Configuring proxy settings...
Found vagrant-proxyconf plugin - Now checking envinronent variables...
http_proxy is set to: http://proxy:80
Bringing machine 'connection' up with 'virtualbox' provider...
==> connection: Box 'bento/ubuntu-16.04' could not be found. Attempting to find and install...
    connection: Box Provider: virtualbox
    connection: Box Version: >= 0
==> connection: Loading metadata for box 'bento/ubuntu-16.04'
    connection: URL: https://vagrantcloud.com/bento/ubuntu-16.04
==> connection: Adding box 'bento/ubuntu-16.04' (v201708.22.0) for provider: virtualbox
    connection: Downloading: https://vagrantcloud.com/bento/boxes/ubuntu-16.04/versions/201708.22.0/providers/virtualbox.box
==> connection: Successfully added box 'bento/ubuntu-16.04' (v201708.22.0) for 'virtualbox'!
==> connection: Importing base box 'bento/ubuntu-16.04'...
==> connection: Matching MAC address for NAT networking...
==> connection: Checking if box 'bento/ubuntu-16.04' is up to date...
==> connection: Setting the name of the VM: nmos-device-connection-management-ri_connection_1505752809917_71011
==> connection: Clearing any previously set network interfaces...
==> connection: Preparing network interfaces based on configuration...
    connection: Adapter 1: nat
==> connection: Forwarding ports...
    connection: 80 (guest) => 8080 (host) (adapter 1)
    connection: 8858 (guest) => 8858 (host) (adapter 1)
   connection: 8860 (guest) => 8860 (host) (adapter 1)
    connection: 22 (guest) => 2222 (host) (adapter 1)
==> connection: Booting VM...
==> connection: Waiting for machine to boot. This may take a few minutes...
    connection: SSH address: 127.0.0.1:2222
    connection: SSH username: vagrant
    connection: SSH auth method: private key
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
...
```

We configured VRDE in VirtualBox to watch the machine boot. After the GRUB menu shows, we are only faced with a blinking cursor in the top-left of the console.

Can you advise on what the next steps might be?

**Comment by simonrankine** [comment]
Hi Andrew,

Thanks for your message. Apologies for not getting back to you sooner - today is my first day back in the office post IBC.

It looks to me that there is a pretty serious issue with your installation of vagrant - vagrant is unable to SSH into the virtual machine, and without that it can't even begin to start deploying things.

> You appear to be running Vagrant outside of the official installers.
> Note that the installers are what ensure that Vagrant has all required
> dependencies, and Vagrant assumes that these dependencies exist. By
> running outside of the installer environment, Vagrant may not function
> properly. To remove this warning, install Vagrant using one of the
> official packages from vagrantup.com.

Which I'm afraid is a message I've never seen before. Where did you install Vagrant from?

The other thing that concerns me slightly is that you appear to be running vagrant as root. Is there any particular reason for this? From a security point of view it's far from ideal, and it's not something I've tried doing but it could cause a range of unexpected behaviours. I think the first step is to make sure you can get vagrant to spin up a bog standard VM, then try again with CM. For example make sure you can do:

```
$ vagrant init hashicorp/precise64
$ vagrant up
```

Hope that helps,

Regards,

Simon

**Comment by quarker** [comment]
Hi Simon,

Thank you for your response. We found that the root cause was due to nested virtualization as we tried to run a VirtualBox inside a virtual machine. In this case, we were running on an AWS EC2 instance. The nested virtualization error was `VT-x is not available. (VERR_VMX_NO_VMX)` from VirtualBox. As we ran vagrant on local machines, the error disappeared.

In order to get around this issue, we decided not to use VirtualBox and Vagrant, but ran the provision script straight inside the EC2 instance. Here are the steps to get it working:
<code>
    1  cd /
    2  wget https://github.com/bbc/nmos-device-connection-management-ri/archive/master.zip
    3  apt install unzip
    4  unzip master.zip
    5  mv nmos-device-connection-management-ri-master/ vagrant
    6  chown -R ubuntu:ubuntu vagrant/
    7  ./provision_node.sh
</code>
